documentation/services_master_puppetserver.markdown

```diff
@@ -78,11 +72,11 @@ Right now, the main administrative task is forcing expiration of all environment
 
 ### JRuby Interpreters
 
-Most of Puppet Server's work --- compiling catalogs, receiving reports, etc. --- is still done by the Ruby-based Puppet master application. But instead of using the operating system's MRI Ruby runtime, Puppet Server runs Puppet in JRuby, drawing from a separate collection of Ruby code.
+Most of Puppet Server's work — compiling catalogs, receiving reports, etc. — is still done by Ruby code. But instead of using the operating system's MRI Ruby runtime, Puppet Server runs Puppet in JRuby, drawing from a separate collection of Ruby code.
```

Comment:
I don't understand what "drawing from a separate collection of Ruby code" is supposed to mean (I know you didn't write it, but while we're here).

Reply:
Maybe it means "different implementation of Ruby"? I think that last clause could either simply be dropped or give a brief explanation of what Jruby is, something like "...in JRuby, an implementation of the Ruby interpreter that runs on the JVM."

Reply:
Yeah, I could see "collection of Ruby code" meaning Ruby interpreter, though I could just as easily see it meaning different Ruby based Puppet code.